crates/rspack_plugin_javascript/src/visitors/dependency/harmony_import_dependency_scanner.rs

```diff
@@ -432,18 +432,70 @@ impl Visit for HarmonyImportRefDependencyScanner<'_> {
     }
   }
 
+  fn visit_opt_chain_expr(&mut self, opt_chain_expr: &OptChainExpr) {
+    let expression_info = extract_member_expression_chain(opt_chain_expr);
+    // dbg!(&expression_info);
+    let member_chain = expression_info.members();
+    if member_chain.len() > 1
+      && let Some(reference) = self.import_map.get(&member_chain[0])
+    {
+      let mut non_optional_members = expression_info.non_optional_part();
+      // dbg!(&non_optional_members);
+      let start: u32 = opt_chain_expr.span.real_lo();
+      let end = if !non_optional_members.is_empty()
+        && let Some(span) = expression_info
+          .members_spans()
+          .get(non_optional_members.len() - 1)
+      {
+        span.real_hi()
+      } else {
+        opt_chain_expr.span.real_hi()
+      };
+      non_optional_members.pop_front();
+      let mut ids = reference.names.clone().map(|f| vec![f]).unwrap_or_default();
+      ids.extend_from_slice(
+        &non_optional_members
+          .into_iter()
+          .map(|item| item.0.clone())
+          .collect::<Vec<_>>(),
+      );
+      self
+        .rewrite_usage_span
+        .insert(opt_chain_expr.span, ExtraSpanInfo::ReWriteUsedByExports);
```

Comment:
@IWANABETHATGUY Would you please check if this is correct?

Reply:
https://github.com/web-infra-dev/rspack/blob/4f3c181a6e8ddcae7b830956f232d0a5b701c062/crates/rspack_plugin_javascript/src/plugin/inner_graph_plugin.rs#L113-L126, you can refer the related member_expr visitor in `inner_graph_plugin`